packages/edit-site/src/components/add-new-template/utils.js

```diff
@@ -175,184 +86,296 @@ const usePublicTaxonomies = () => {
 	}, [ taxonomies ] );
 };
 
-/**
- * `category` and `post_tag` are handled specifically in template
- * hierarchy so we need to differentiate them and return the rest,
- * e.g. `category-$slug` and `taxonomy-$taxonomy-$term`.
- */
-export const useTaxonomies = () => {
-	const taxonomies = usePublicTaxonomies();
-	const specialTaxonomies = [ 'category', 'post_tag' ];
-	return useMemo(
+export const usePostTypeMenuItems = ( onClickMenuItem ) => {
+	const publicPostTypes = usePublicPostTypes();
+	const existingTemplates = useExistingTemplates();
+	const defaultTemplateTypes = useDefaultTemplateTypes();
+	// `page`is a special case in template hierarchy.
+	const templatePrefixes = useMemo(
 		() =>
-			taxonomies?.filter(
-				( { slug } ) => ! specialTaxonomies.includes( slug )
-			),
-		[ taxonomies ]
+			publicPostTypes?.reduce( ( accumulator, { slug } ) => {
+				let suffix = slug;
+				if ( slug !== 'page' ) {
+					suffix = `single-${ suffix }`;
+				}
+				accumulator[ slug ] = suffix;
+				return accumulator;
+			}, {} ),
+		[ publicPostTypes ]
 	);
-};
-
-export const useTaxonomyCategory = () => {
-	const taxonomies = usePublicTaxonomies();
-	return useMemo(
-		() => taxonomies?.filter( ( { slug } ) => slug === 'category' ),
-		[ taxonomies ]
+	// We need to keep track of naming conflicts. If a conflict
+	// occurs, we need to add slug.
+	const postTypeLabels = publicPostTypes?.reduce(
+		( accumulator, { labels } ) => {
+			const singularName = labels.singular_name.toLowerCase();
```

Comment:
It seems singular_name label is used when one is dealing with a specific item inside a screen just for that specific taxonomy so "Product categories" singular_name is just "Category". But taxonomies also have a human readable name when used in general WordPress UI, and for example, on WooCommerce categories it is "Product categories".
![image](https://user-images.githubusercontent.com/11271197/179211707-85398b8a-fee7-40bc-98dd-7b862e4ec2d8.png)


Could we rely on the name label? I guess for most cases it would avoid the need to have the slug. Things would be named on the plural but I think that's ok.

Reply:
I don't think we can because it's not guaranteed that the `name` is unique - it even defaults to [categories/tags](https://developer.wordpress.org/reference/functions/get_taxonomy_labels/).